LPD-43273 Add necessary style book entry service methods for page specification endpoints

lfbesada opened the pull request:
Hi Team,

As part of implementing [Allow Basic CRUD operations for Content Pages via Headless APIs](https://liferay.atlassian.net/browse/LPD-22802) in Page Management, we need StyleBookEntry services to fetch data by external reference code.
This PR adds the necessary services and integration tests.
Could you please review it? 
Thanks in advance! :heart:

liferay-continuous-integration commented:
The following guidelines have been set by the owner of this repository: 
- &nbsp;&nbsp;&nbsp;&nbsp;"Thank you for sending this PR to liferay-platform-experience. If you haven't already,take this opportunity to review our [pull request guidelines](https://liferay.atlassian.net/wiki/spaces/PEDS/pages/2915598365). For additional help,please reach out via the [appropriate channels](https://liferay.atlassian.net/wiki/spaces/PEDS/overview#%3Aslack%3A-Slack-Channels)."

lfbesada commented:
ci:test:sf

lfbesada commented:
ci:test:relevant

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-6.liferay.com/job/test-portal-source-format/7620/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-6.liferay.com/userContent/jobs/test-portal-source-format/builds/7620/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 5 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/273c02f8c36f1cc973b677ca7754ed345a06fb90">273c02f8c36f1cc973b677ca7754ed345a06fb90</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/lfbesada/liferay-portal/tree/LPD-43273-Add-methods-fetch-style-book-by-external-reference-code">LPD-43273-Add-methods-fetch-style-book-by-external-reference-code</a><br/>Branch GIT ID: <a href="https://github.com/lfbesada/liferay-portal/commit/d0b0ba4bc6d7c9d84bc628a17d45c4a1603a80cb">d0b0ba4bc6d7c9d84bc628a17d45c4a1603a80cb</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-6.liferay.com/job/test-portal-source-format/7620/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-6.liferay.com/userContent/jobs/test-portal-source-format/builds/7620/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:relevant - 0 out of 1 jobs passed in 1 hour 56 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/273c02f8c36f1cc973b677ca7754ed345a06fb90">273c02f8c36f1cc973b677ca7754ed345a06fb90</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/aa943208fe01a0f0630284b8f7fb301162cd300d">aa943208fe01a0f0630284b8f7fb301162cd300d</a><br/>Jenkins Build URL: <a href="https://testray.liferay.com/#/project/35392/routines/590307/build/81178721">EE Development Acceptance (master) - 1054 - 2024-11-29[16:36:01]</a></p><details><summary><strong>ci:test:relevant - 0 out of 1 jobs PASSED</strong></summary><div><h4>1 Failed Jobs:</h4><ul><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest(master)/11734/"><strike>test-portal-acceptance-pullrequest(master)</strike></a></li></ul></div></details><h5>For more details click <a href="https://test-1-31.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/11734/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest(master)/11734/">test-portal-acceptance-pullrequest(master)</a></h5><div><h6>Job Results:</h6><p>0 Jobs Passed.<br/>1 Job Failed.</p></div><pre>[beanshell]      [exec] * What went wrong:
[beanshell]      [exec] Execution failed for task ':yarnInstall'.
[beanshell]      [exec] &gt; Process 'command '/opt/dev/projects/github/liferay-portal/build/node/bin/node'' finished with non-zero exit value 1
[beanshell]      [exec] 
[beanshell]      [exec] * Try:
[beanshell]      [exec] &gt; Run with --info or --debug option to get more log output.
[beanshell]      [exec] &gt; Run with --scan to get full insights.
[beanshell]      [exec] &gt; Get more help at https://help.gradle.org.
[beanshell]      [exec] 
[beanshell]      [exec] * Exception is:
[beanshell]      [exec] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':yarnInstall'.
[beanshell]      [exec]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda\$executeIfValid\$1(ExecuteActionsTaskExecuter.java:148)
[beanshell]      [exec]   at org.gradle.internal.Try\$Failure.ifSuccessfulOrElse(Try.java:282)
[beanshell]      [exec]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:146)
[beanshell]      [exec]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:134)
[beanshell]      [exec]   at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
[beanshell]      [exec]   at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
[beanshell]      [exec]   at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
[beanshell]      [exec]   at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
[beanshell]      [exec]   at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
[beanshell]      [exec]   at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter\$1.executeTask(EventFiringTaskExecuter.java:77)
[beanshell]      [exec]   at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter\$1.call(EventFiringTaskExecuter.java:55)
[beanshell]      [exec]   at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter\$1.call(EventFiringTaskExecuter.java:52)</pre></div></li></ol><h4>For upstream results, click <a href="https://test-1-40.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">here</a>.</h4></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://test-1-31.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/11734/bundles/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a></li></ul>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest(master)/11734/">test-portal-acceptance-pullrequest(master)#11734</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-31.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/11734/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>relevant<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-platform-experience/liferay-portal/pull/829">liferay-platform-experience#829</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/81705155">[master] ci:test:relevant - lfbesada &gt; liferay-platform-experience - PR#829 - 2024-12-02[06:39:40]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>81705155<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-31.liferay.com/job/publish-testray-report/27037/">publish-testray-report#27037</a><br/></div></div>

algobob commented on modules/apps/style-book/style-book-api/src/main/resources/com/liferay/style/book/service/packageinfo:
```diff
@@ -1 +1 @@
-version 5.0.0
```
Usually we create a "semantic version" commit  for `packageinfo` and `bnd` files version changes only :slightly_smiling_face:

lfbesada replied:
@algobob Thanks for the review! ❤️ 
Should I split the auto-generated code by buildService into two distinct commits?
Thanks!

thiago-buarqque replied:
Please @lfbesada, merge these changes with the `buildService` commit. They are auto-generated by buildService.

algobob commented on modules/apps/style-book/style-book-test/src/testIntegration/java/com/liferay/style/book/service/test/StyleBookEntryServiceTest.java:
```diff
@@ -121,6 +123,47 @@ public void testDeleteStyleBookEntryByExternalReferenceCodeWithoutDeletePermissi
+			styleBookEntry.getStyleBookEntryId(),
+			curStyleBookEntry.getStyleBookEntryId());
+
+		_removeResourcePermissions(styleBookEntry.getStyleBookEntryId());
+
+		try {
+			UserTestUtil.setUser(
+				UserTestUtil.addGroupUser(_group, RoleConstants.SITE_MEMBER));
+
+			_styleBookEntryService.fetchStyleBookEntryByExternalReferenceCode(
+				styleBookEntry.getExternalReferenceCode(),
+				styleBookEntry.getGroupId());
+
+			Assert.fail();
+		}
+		catch (PrincipalException principalException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(principalException);
+			}
+		}
+		finally {
+			UserTestUtil.setUser(TestPropsValues.getUser());
+		}
```
I think you can create another test for the permission part :thinking: like the `testGetStyleBookEntryByExternalReferenceCode` one is doing

algobob replied:
Oh, you did in the next commit. Try to re-order and keep the performance improvement commit as the first one for the tests part, so it is already clear about this part :slightly_smiling_face:

algobob commented on modules/apps/style-book/style-book-test/src/testIntegration/java/com/liferay/style/book/service/test/StyleBookEntryServiceTest.java:
```diff
@@ -151,16 +194,7 @@ public void testGetStyleBookEntryByExternalReferenceCodeWithoutViewPermission()
-			ResourceConstants.SCOPE_INDIVIDUAL,
-			String.valueOf(styleBookEntry.getStyleBookEntryId()),
-			ActionKeys.VIEW);
+		_removeResourcePermissions(styleBookEntry.getStyleBookEntryId());
```
Usually if there is only one usage, we keep it into the test method, but since it was extracted to a method, could you also apply it in `testGetStyleBookEntryByExternalReferenceCodeWithoutViewPermission` ? It seems same code is also there.

algobob replied:
Oh, I just see your next commit about improve performance. Since your new test was already with this improvement, to make easier to understand the flow, re-order the commit :slightly_smiling_face:

algobob commented on modules/apps/style-book/style-book-test/src/testIntegration/java/com/liferay/style/book/service/test/StyleBookEntryServiceTest.java:
```diff
@@ -151,16 +194,7 @@ public void testGetStyleBookEntryByExternalReferenceCodeWithoutViewPermission()
-			ResourceConstants.SCOPE_INDIVIDUAL,
-			String.valueOf(styleBookEntry.getStyleBookEntryId()),
-			ActionKeys.VIEW);
+		_removeResourcePermissions(styleBookEntry.getStyleBookEntryId());
```
Since you extracted this part, it seems `testGetStyleBookEntryByExternalReferenceCodeWithoutViewPermission` also have a duplication. Could you replace it for this new method?

algobob replied:
Also, please rename this commit. SF does not change logic.

thiago-buarqque replied:
@lfbesada Since you're refactoring, could you please create a setup for

```java
StyleBookEntry styleBookEntry =
  _styleBookEntryService.addStyleBookEntry(
    RandomTestUtil.randomString(), _group.getGroupId(),
    RandomTestUtil.randomString(), null,
    RandomTestUtil.randomString(), _serviceContext);
```

and other code duplication please?

algobob commented:
I just added some comments :slightly_smiling_face: 

Also, SF commits usually only re-order/re-indent code. In case you refactor code by rename/delete/move out, Its better to create another commit for it

lfbesada commented:
ci:test:sf

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-37.liferay.com/job/test-portal-source-format/10766/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-source-format/builds/10766/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 13 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/c163f5a502d7faee3fc9eb8dce30f2b876da3d5f">c163f5a502d7faee3fc9eb8dce30f2b876da3d5f</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/lfbesada/liferay-portal/tree/LPD-43273-Add-methods-fetch-style-book-by-external-reference-code">LPD-43273-Add-methods-fetch-style-book-by-external-reference-code</a><br/>Branch GIT ID: <a href="https://github.com/lfbesada/liferay-portal/commit/5ba8521ce6bbfac69297d02a01c7c3d6b6c12be2">5ba8521ce6bbfac69297d02a01c7c3d6b6c12be2</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-37.liferay.com/job/test-portal-source-format/10766/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-source-format/builds/10766/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-37.liferay.com/job/test-portal-source-format/10766/">test-portal-source-format#10766</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-source-format/builds/10766/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-platform-experience/liferay-portal/pull/829">liferay-platform-experience#829</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/82544703">[master] ci:test:sf - lfbesada &gt; liferay-platform-experience - PR#829 - 2024-12-03[02:22:10]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>82544703<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-37.liferay.com/job/publish-testray-report/40982/">publish-testray-report#40982</a><br/></div></div>

algobob commented on modules/apps/style-book/style-book-test/src/testIntegration/java/com/liferay/style/book/service/test/StyleBookEntryServiceTest.java:
```diff
@@ -71,6 +71,9 @@ public void testAddStyleBookEntryWithoutAddPermission() throws Exception {
 			Assert.fail();
 		}
 		catch (PrincipalException principalException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(principalException);
+			}
```
SF does not change logic. Please rename this commit.

lfbesada commented:
Hi @algobob,
Thanks for the review!
Could you please elaborate on the specific logic changes you're seeing in these changes?
I'm having trouble seeing how I've altered the logic, and I'm unsure why you'd like me to rename the commits. From my perspective, the changes are primarily related to source formatting:

1. [LPD-43273 SF join methods to improve performance](https://github.com/liferay-platform-experience/liferay-portal/pull/829/commits/10df856922e6a26eb9980cbdf762d09812b3b09e) I've merged test methods that cover the same functionality under different circumstances, following Brian's pattern as seen [here](https://github.com/brianchandotcom/liferay-portal/pull/156308#issuecomment-2488644048)
2. [LPD-43273 SF extract and reuse](https://github.com/liferay-platform-experience/liferay-portal/pull/829/commits/f92622d39d152310c2920d49c6f7c5085fe1679b) I've extracted duplicate code into a separate method without changing the underlying logic.
3. [LPD-43273 SF](https://github.com/liferay-platform-experience/liferay-portal/pull/829/commits/acbd9ec6acc7a169f91644a4730add3d5d5af23a) I've added debugging traces for exceptions to provide more context during test debugging, as requested by Brian in the past, for example [here](https://github.com/brianchandotcom/liferay-portal/pull/153180#issuecomment-2283727857)

Thanks in advance for your clarification!

Best Regards

lfbesada commented:
Hi @algobob ,
I've revised the commit messages for the three commits to remove the "SF" references and provide more context. Could you please take a look and let me know if they're more suitable now? Thanks!

algobob commented:
> Hi @algobob, Thanks for the review! Could you please elaborate on the specific logic changes you're seeing in these changes? I'm having trouble seeing how I've altered the logic, and I'm unsure why you'd like me to rename the commits. From my perspective, the changes are primarily related to source formatting:
> 
> 1. [LPD-43273 SF join methods to improve performance](https://github.com/liferay-platform-experience/liferay-portal/pull/829/commits/10df856922e6a26eb9980cbdf762d09812b3b09e) I've merged test methods that cover the same functionality under different circumstances, following Brian's pattern as seen [here](https://github.com/brianchandotcom/liferay-portal/pull/156308#issuecomment-2488644048)
> 2. [LPD-43273 SF extract and reuse](https://github.com/liferay-platform-experience/liferay-portal/pull/829/commits/f92622d39d152310c2920d49c6f7c5085fe1679b) I've extracted duplicate code into a separate method without changing the underlying logic.
> 3. [LPD-43273 SF](https://github.com/liferay-platform-experience/liferay-portal/pull/829/commits/acbd9ec6acc7a169f91644a4730add3d5d5af23a) I've added debugging traces for exceptions to provide more context during test debugging, as requested by Brian in the past, for example [here](https://github.com/brianchandotcom/liferay-portal/pull/153180#issuecomment-2283727857)
> 
> Thanks in advance for your clarification!
> 
> Best Regards

Hi @lfbesada 

Sure! By SF, I mean changes made by source format command `gw formatSource`, which basically consists in rename/reorder variables to match patterns and reorder methods.

When we refactor code, like extracting methods & moving out codes to reuse and avoid duplication, it's better to make a different commit to avoid confusion related with SF code changes.

Also, when Brian ask some changes, sometimes it makes sense to be same commit as SF, but sometimes it can be a different commit. In this case, since it were changes to add logs and to join methods, it seems better to be a different commit :slightly_smiling_face:  

Makes sense?

algobob commented:
> Hi @algobob , I've revised the commit messages for the three commits to remove the "SF" references and provide more context. Could you please take a look and let me know if they're more suitable now? Thanks!

Sure! It's more suitable now :slightly_smiling_face:

algobob commented:
ci:test:sf

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-33.liferay.com/job/test-portal-source-format/3776/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-33.liferay.com/userContent/jobs/test-portal-source-format/builds/3776/jenkins-report.html">here</a>.</b></p>

lfbesada commented:
> Sure! By SF, I mean changes made by source format command `gw formatSource`, which basically consists in rename/reorder variables to match patterns and reorder methods.
> 
> When we refactor code, like extracting methods & moving out codes to reuse and avoid duplication, it's better to make a different commit to avoid confusion related with SF code changes.
> 
> Also, when Brian ask some changes, sometimes it makes sense to be same commit as SF, but sometimes it can be a different commit. In this case, since it were changes to add logs and to join methods, it seems better to be a different commit 🙂
> 
> Makes sense?

I understand your perspective, but I believe that any code change should include the result of gw formatSource in the same commit not in a separate one. This makes it much easier to track changes and ensures that our codebase always reflects the correct formatting.
Personally, I only create commits with "SF Auto" messages when the formatting in the master branch is incorrect. However, when working on older classes, I try to follow the "Boy Scout rule" and leave the code better than I found it. So, if I see opportunities to improve source formatting based on our current standards (not just auto-generated changes), I'll include "SF" in the commit message to clarify that it's a style change and not a logical change.
It's just a difference in approach :raised_hands: 
Thanks for your review!

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 16 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/c163f5a502d7faee3fc9eb8dce30f2b876da3d5f">c163f5a502d7faee3fc9eb8dce30f2b876da3d5f</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/lfbesada/liferay-portal/tree/LPD-43273-Add-methods-fetch-style-book-by-external-reference-code">LPD-43273-Add-methods-fetch-style-book-by-external-reference-code</a><br/>Branch GIT ID: <a href="https://github.com/lfbesada/liferay-portal/commit/468cf52c6213211dbb7963aa3932dcb20d07d7d1">468cf52c6213211dbb7963aa3932dcb20d07d7d1</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-33.liferay.com/job/test-portal-source-format/3776/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-33.liferay.com/userContent/jobs/test-portal-source-format/builds/3776/jenkins-report.html">here</a>.</h5></details></html>

algobob commented:
> > Sure! By SF, I mean changes made by source format command `gw formatSource`, which basically consists in rename/reorder variables to match patterns and reorder methods.
> > When we refactor code, like extracting methods & moving out codes to reuse and avoid duplication, it's better to make a different commit to avoid confusion related with SF code changes.
> > Also, when Brian ask some changes, sometimes it makes sense to be same commit as SF, but sometimes it can be a different commit. In this case, since it were changes to add logs and to join methods, it seems better to be a different commit 🙂
> > Makes sense?
> 
> I understand your perspective, but I believe that any code change should include the result of gw formatSource in the same commit not in a separate one. This makes it much easier to track changes and ensures that our codebase always reflects the correct formatting. Personally, I only create commits with "SF Auto" messages when the formatting in the master branch is incorrect. However, when working on older classes, I try to follow the "Boy Scout rule" and leave the code better than I found it. So, if I see opportunities to improve source formatting based on our current standards (not just auto-generated changes), I'll include "SF" in the commit message to clarify that it's a style change and not a logical change. It's just a difference in approach 🙌 Thanks for your review!

Oh, I think I didn't explain in a good way my perspective and made I confusion :brain: :exploding_head: 

You are completely right when you say to run gw formatSource with every commit and always improve current. That's the way to go. We should not run it as a last commit. What I got confused was about the "SF" in the commit message. Usually when I see "SF" commits, it's just for formatting like the "SF Auto" you mentioned.

In the end, we have the same perspective. You used "SF" for your improvements, but we use to just mention "refactor" or something similar. That was the my confusion :slightly_smiling_face:

algobob commented:
LGTM :rocket: 

cc: @ethib137 @thiago-buarqque Let us know if you have any comments.

Also, should this PR be sent to bpm as well?

ealonso commented:
Hey @algobob,

Why do you think this pr should send to bpm? it is only changing logic in style books and style books is part of your components, nothing related with bpm.

algobob commented:
> Hey @algobob,
> 
> Why do you think this pr should send to bpm? it is only changing logic in style books and style books is part of your components, nothing related with bpm.

Hey @ealonso 

Currently, we dont have a backend senior developer with long experience in Liferay-platform in our team, so as Brian requested, it's part of our process to send our PR's to bpm to review as well. Since this is a external PR, I dont know if we should follow same process, so that's why I'm double-checking :slightly_smiling_face:

ealonso commented:
@algobob Ok, thanks for letting me know.

In case it helps I've already reviewed it and it looks good to me, also, Lourdes is a Staff Backend Developer, so I think we can trust in her code, hehe.

thiago-buarqque commented on modules/apps/style-book/style-book-test/src/testIntegration/java/com/liferay/style/book/service/test/StyleBookEntryLocalServiceTest.java:
```diff
@@ -99,6 +99,26 @@ public void testDeleteStyleBookEntryByExternalReferenceCode()
+				fetchStyleBookEntryByExternalReferenceCode(
+					externalReferenceCode, _group.getGroupId()));
+
+		_styleBookEntryLocalService.getStyleBookEntryByExternalReferenceCode(
+			externalReferenceCode, _group.getGroupId());
```
@lfbesada  You forgot to assert that the `StyleBookEntry` is not null.

Also, since we have both **get** and **fech**, please assert that an exception is thrown in case there is no StyleBookEntry and `getStyleBookEntryByExternalReferenceCode` is called. And null is returned in case `fetchStyleBookEntryByExternalReferenceCode` is called

ealonso replied:
She did in that way because Brian asked for it in other prs

thiago-buarqque replied:
@ealonso what did he say? Just "hey just do it because I am asking to"?

ealonso replied:
I'm trying to find the commit, to clarify why we are not assert for null

thiago-buarqque reviewed (changes requested):
Thanks @lfbesada. I'd like to know why do you need both **_get_** and **_fetch_** by ERC

ealonso commented:
Hey @ethib137,

The review of this pr is a joke, right?, It is not make any sense the comments of @thiago-buarqque and @algobob, please help me to understand it because for me it is not makes any sense.

@algobob is asking for changes, and @thiago-buarqque is asking for the opposite ...

ealonso commented:
@ethib137 @thiago-buarqque @algobob,

We are trying to contribute some code, that should be develop by your team, and you are doing it really difficult. Your reviews and your comments don't make any sense.

So, at this point I'm going to close the pr, and I'm going to talk with EPM and PM teams and escale it properly.

Regards,
Eudaldo.

ealonso commented:
ci:close

thiago-buarqque commented:
@ealonso 

>@algobob is asking for changes, and @thiago-buarqque is asking for the opposite ...

You are probably saying this because of 

https://github.com/liferay-platform-experience/liferay-portal/pull/829#discussion_r1866496420

In there @algobob says

> In this case, should this version change be same commit or in a different commit? @thiago-buarqque @ethib137

He is unsure, so he asked, and I answered the opposite of what he said, he just made a mistake

thiago-buarqque commented:
@ealonso what is not good exactly? You just said "this is a joke" and didn't mention what's not good. Neither answered why Brian asked to add both fetch and get methods. 

If you really don't want to go with this, talk with our PM and we'll work on this when feasible.

Thanks

ealonso commented:
@thiago-buarqque 

https://github.com/liferay-platform-experience/liferay-portal/pull/829#discussion_r1866430222 and https://github.com/liferay-platform-experience/liferay-portal/pull/829#discussion_r1868397963

https://github.com/liferay-platform-experience/liferay-portal/pull/829#issuecomment-2515176962 and see this commit in this pr https://github.com/liferay-platform-experience/liferay-portal/pull/796/commits/b4c4395592403d32727cdefecd684bdcf3a13191

thiago-buarqque commented:
@ealonso 

Regarding https://github.com/liferay-platform-experience/liferay-portal/pull/829#discussion_r1866430222 @algobob [then says](https://github.com/liferay-platform-experience/liferay-portal/pull/829#discussion_r1866496420)

> I think its better to keep auto-generated code in one commit. But now I have a doubt. This version change was also auto-generated, right? In this case, should this version change be same commit or in a different commit?

So he made a mistake. And I answered in [here](https://github.com/liferay-platform-experience/liferay-portal/pull/829#discussion_r1868397963). Don't see the big issue with that

thiago-buarqque commented:
@ealonso regarding https://github.com/liferay-platform-experience/liferay-portal/commit/b4c4395592403d32727cdefecd684bdcf3a13191, yeah, you are right, he made renaming in that commit, something that should not be there. But you should use that to say

> Hey, you're saying [this](https://github.com/liferay-platform-experience/liferay-portal/pull/829#issuecomment-2515176962), but then does the opposite [here](https://github.com/liferay-platform-experience/liferay-portal/commit/b4c4395592403d32727cdefecd684bdcf3a13191).

If I did not ask, I'd never see that commit, I am not in the other PR (I did not review it there yet and when I do, I will ask him to modify that commit)

algobob commented:
Easy, team :pray: 

Remember pull requests reviews is a space for learning as well.

@ealonso About https://github.com/liferay-platform-experience/liferay-portal/pull/829#discussion_r1866430222 , the conversation was still open. @lfbesada asked if she should split in two commits. I made a suggestion, but tagged my team for confirmation yet. 

About the https://github.com/liferay-platform-experience/liferay-portal/commit/b4c4395592403d32727cdefecd684bdcf3a13191 , I've seen this kind of commit before in Liferay.  I think its not an big issue, but I agree with your comment there that is better to run SF for every commit to make easier to read, so I'm working on the changes.

ealonso commented:
Hey @thiago-buarqque,

I'm complaining, because we send a pr that adds two new methods in a service builder and we spend some time improving your code. IMHO, it's an easy pr to review, and I said Lourdes please send it to Platform Experience for further review.

Then the review for your team was: someone in your team asked for changes (In my opinion some of those changes doesn't make any sense but we did it because it is your code) we did all the changes and redo de pr to match with your requirements (it consume time). After that, another person review the pr and asked for oposite changes, also the person who review the pr don't follow the pattern asked in this pr in his pr ...  So, for me it is like what? seriously?

And the worst thing is that all of it is for a simple pr that adds two new methods in a ServiceBuilder class. 

Next time I think it is better if we asked for those methods and let the owner team implement the code, actually I'm talking with your PM and EPM, so they can properly prioritize it and you can implement it in the way that you want, we are not going to invest more time in this pr.

Regards,
Eudaldo.
